mpa/seg/inferrer.py

```diff
@@ -108,16 +109,26 @@ def infer(self, cfg):
             _ = load_checkpoint(model, cfg.load_from, map_location='cpu')
 
         # Inference
-        model = MMDataParallel(model, device_ids=[0])
+        model.eval()
+        eval_model = MMDataParallel(model, device_ids=[0])
 
         # InferenceProgressCallback (Time Monitor enable into Infer task)
-        SegStage.set_inference_progress_callback(model, cfg)
-
-        segmentations = single_gpu_test(model, data_loader, output_logits=True)
+        SegStage.set_inference_progress_callback(eval_model, cfg)
+
+        segmentations = []
+        feature_vectors = []
+        for data in mm_val_dataloader:
+            with torch.no_grad():
+                result, feature_vector = eval_model(return_loss=False, output_logits=True, **data)
+                segmentations.append(result)
+                feature_vectors.append(feature_vector)
+            assert len(result) == 1
```

Comment:
- Let's avoid inserting assert in product code; explicitly raise exception.
- Simple loop looks concise, but it will lose batch_size>1 execution. single_gpu_test() seems support it.
- In addition, if you perform grep 'return_repr_vector' you would find some matches; could you check if the 'feature vector' functionality is already implemented in ote mmseg?